### PR TITLE
Add Support to Get Image in Registration Form

### PR DESCRIPTION
**Pull request by AKSHILMY**
image

### CODE REVIEW

**Comment by Ishad-M-I-M** [comment]
There's a discussion on storing images in the database itself instead in Azure Blob storage.

**Comment by AKSHILMY** [comment]
> There's a discussion on storing images in the database itself instead in Azure Blob storage.

@achiraDias Any modifications made up-to now for this requirement?

File: controllers/student.controller.js [hunk 1]
```diff
@@ -2,169 +2,116 @@ const { studentFormValidation } = require("./../validation/form");
...
-        .catch((error) => {
-          console.error("Failed to retrieve data: ", error);
-          res.status(500).json({ error: "Failed to retrieve data" });
-        });
-    })
-    .catch((error) => {
-      console.error("Something went wrong: ", error.message);
-      res.status(500).json({ error: "Something went wrong" });
-    });
+    let {index, email, firstName, lastName, faculty, department} = req.body;
```
Comment (Ishad-M-I-M) [inline controllers/student.controller.js]:
```suggestion
```

File: controllers/student.controller.js [hunk 2]
```diff
@@ -2,169 +2,116 @@ const { studentFormValidation } = require("./../validation/form");
...
+                    console.log(results);
+                    res.status(200).json(results);
+                })
+                .catch((error) => {
+                    res.status(500).json({ error: error.message });
+                });
         })
         .catch((error) => {
-          res.status(500).json({ error: error.message });
+            console.error("SOMETHING WRONG : ", error.message);
```
Comment (Ishad-M-I-M) [inline controllers/student.controller.js:114]:
```suggestion
            console.error("SOMETHING WRONG : ", error.message);
            res.status(500).json({ error: error.message });
```

File: uploads/storage.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const filePathGenerator = (req,file) => {
+    // Generate a unique filename for each uploaded file
+    const uniqueSuffix = req.body.index;
+    const fileExtension = file.originalname.split('.').pop();
+    return file.fieldname + '-' + uniqueSuffix + '.' + fileExtension
+}
+
+module.exports = {
+    multer,storage,upload,removeFile,filePathGenerator
+  };
```
Comment (Ishad-M-I-M) [inline uploads/storage.js]:
```suggestion
  };
```
```suggestion
  };
  
```

**Pull request merged**